src/Juvix/Compiler/Nockma/Evaluator/Storage.hs

```diff
@@ -24,14 +24,28 @@ stripTags = \case
           $ c
       )
 
-instance (Eq a) => Eq (StorageKey a) where
-  s1 == s2 = stripTags (s1 ^. storageKeyTerm) == stripTags (s2 ^. storageKeyTerm)
+instance (NockmaEq a) => NockmaEq (StorageKey a) where
+  nockmaEq (StorageKey s1) (StorageKey s2) = nockmaEq s1 s2
 
-instance (Hashable a) => Hashable (StorageKey a) where
-  hashWithSalt s k = hashWithSalt s (stripTags (k ^. storageKeyTerm))
+instance (NockmaEq a) => Eq (StorageKey a) where
+  (==) = nockmaEq
+
+instance (NockmaEq a, Hashable a) => Hashable (StorageKey a) where
+  hashWithSalt salt k = goTerm salt (k ^. storageKeyTerm)
+    where
+      goCell :: Int -> Cell a -> Int
```

Comment:
Does the salt need to be passed in as an argument to each function? Or can the `salt` argument be used directly instead?

Reply:
The top `salt` can be used since the `go` functions are not recursive

Reply:
done